### PR TITLE
Add highlighted dishes to Restaurant show page

### PR DESCRIPTION
**Pull request by AmberWilkie**
PT Story: https://www.pivotaltracker.com/story/show/131070359

Changes proposed in this pull request:
- Creates `highlight` boolean for dishes
- Puts highlight checkbox on dish edit and create pages
- Displays highlighted dishes (max six) on restaurant show pages
- Updates seed to randomly highlight some dishes

What I have learned working on this feature:
- I'm super-crap at design.

Screenshots: 
screen shot 2016-10-08 at 12 22 49 pm

Ready for review!

### CODE REVIEW

**Review by thesuss — COMMENTED** [review]
Why did you move it? Was it in the scope of this feature? :)

File: db/seeds.rb [hunk 1]
```diff
@@ -44,13 +44,19 @@ def build_menus_with_dishes(menu_names)
...
+    highlight = false
+  end
   prefix = restaurant.category
   Dish.create(name: "#{prefix} Dish #{number}",
               price: price,
               description: "A great #{restaurant.category} dish!",
               # category: Dish::CATEGORIES.sample
               menu_ids: menu.id,
-              restaurant_id: restaurant.id
+              restaurant_id: restaurant.id,
```
Comment (tochman) [inline db/seeds.rb]: `restaurant: restaurant` should do. ActiveRecord magic ;-)

File: features/highlight_dish.feature [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    | name   | description     | price |
+    | Pizza  | Delicious pizza | 70    |
+    | Salad  | Leafy           | 65    |
+    | Olives | Salty           | 90    |
+
+Scenario: I add a dish to the highlight list
+  Given I am on the edit dish page for "Pizza"
+  When I check the "Highlight" box
+  And I click the "Submit" button
+  When I am on my restaurant page
```
Comment (tochman) [inline features/highlight_dish.feature]: Shouldn't this be `Then I should be on my restaurant page` ?  It doesn't really make sense the way it is now.

File: features/highlight_dish.feature [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  Then I should see "Pizza"
+  And I should not see "Olives"
+
+Scenario: I view the restaurant page as a visitor
+  Given I am not logged in
+  And "Pizza" is a highlighted dish
+  When I am on my restaurant page
```
Comment (tochman) [inline features/highlight_dish.feature:25]: Same here...

File: features/step_definitions/restaurant_steps.rb [hunk 4]
```diff
@@ -105,6 +105,13 @@
   login_as(user, scope: :user)
 end
 
+Given(/^"([^"]*)" is a highlighted dish$/) do |dish_name|
+  dish = Dish.find_by(name: dish_name)
+  dish.highlight = true
+  dish.save
+  expect(dish.highlight).to eq true
```
Comment (tochman) [inline features/step_definitions/restaurant_steps.rb]: Please don't add assertions in steps that are setting the stage for the test ('Given' steps)

**Comment by AmberWilkie** [comment]
@thesuss I moved it the `_form` partial because I was the one who had moved it to `application` in the first place and it doesn't belong there, according to the pattern of the other folders. But if it has to stay there to be convenient for you, it's zero problem. When I pull upstream right now I'll just keep whatever you have.

File: features/step_definitions/restaurant_steps.rb (hunk 4 above)
Comment (AmberWilkie) [reply features/step_definitions/restaurant_steps.rb]: Right, yes, my bad. I should have taken that out after it worked the first time.

File: features/highlight_dish.feature (hunk 3 above)
Comment (AmberWilkie) [reply features/highlight_dish.feature:25]: Well here it's ok, right? Viewing your restaurant page when not logged in? I was just trying to use steps we already had written.

Comment (tochman) [reply features/highlight_dish.feature:25]: Okay, but it should say `And`

**Comment by tochman** [comment]
Rounded corners is sooo 2013! ;-)

**Comment by thesuss** [comment]
It is. But 2014 ;-)

**Comment by AmberWilkie** [comment]
I gave it a try but obviously I'm not very good at this stuff. Hoping one of the front-enders will rescue me from myself.

File: db/seeds.rb (hunk 1 above)
Comment (AmberWilkie) [reply db/seeds.rb]: done

File: features/highlight_dish.feature (hunk 2 above)
Comment (AmberWilkie) [reply features/highlight_dish.feature]: When
And
And
Then

File: features/highlight_dish.feature (hunk 3 above)
Comment (AmberWilkie) [reply features/highlight_dish.feature:25]: done

**Comment by tochman** [comment]
This feels 'done' IMO. Centering the map on marker click was a suggestion but not part of this story. You have surely learned more than the insight about your design skills when working on this feature @AmberWilkie , haven't you?

**Comment by AmberWilkie** [comment]
This isn't the center-map-markers one :)

File: app/controllers/dishes_controller.rb [hunk 5]
```diff
@@ -37,7 +37,7 @@ def update
   private
 
   def dish_params
-    params.require(:dish).permit(:name, :description, :price, :allergies, :ingredients, :calories, {menu_ids: []})
+    params.require(:dish).permit(:name, :description, :price, :allergies, :ingredients, :calories, {menu_ids: []}, :highlight)
```
Comment (tochman) [inline app/controllers/dishes_controller.rb]: I would add line breaks to avoid loc's longer than 80 charackters

**Review by tochman — CHANGES_REQUESTED** [review]

**Comment by tochman** [comment]
Wow, I'm getting confused by the amount of PRs My bad @AmberWilkie

File: app/controllers/dishes_controller.rb (hunk 5 above)
Comment (AmberWilkie) [reply app/controllers/dishes_controller.rb]: done

**Review by tochman — APPROVED** [review]

**Pull request merged**